tower_cli/cli/transfer/common.py

```diff
@@ -0,0 +1,367 @@
+        'fail_on_multiple_results': False,
+        'all_pages': True
+    })
+
+    # We have to temporarily stash these.
+    # At the end of the process we need to go through all of the nodes and resolve the different
+    # node types from their IDs to their names
+    workflow_nodes_extracted = []
+
+    # This is a stash for us to map the IDs back to the labels
+    workflow_node_to_name_mapping = {}
+
+    node_number = 0
+    for workflow_node in nodes['results']:
+        node_name = 'node{}'.format(node_number)
+        node_number = node_number + 1
+        node_to_add = {
+            "name": node_name,
+        }
+        workflow_node_to_name_mapping[workflow_node['id']] = node_name
+
+        map_node_to_post_options(workflow_node_post_options, workflow_node, node_to_add)
+
+        # We can delete the workflow_job_template since we will be applying it to this workflow
+        del node_to_add["workflow_job_template"]
+
+        # Now we need to resolve the unified job template
+        del node_to_add["unified_job_template"]
+        node_to_add['unified_job_type'] = workflow_node["summary_fields"]["unified_job_template"]["unified_job_type"]
+        node_to_add['unified_job_name'] = workflow_node["summary_fields"]["unified_job_template"]["name"]
+
+        if 'credential' in workflow_node and workflow_node['credential']:
+            node_to_add['credential'] = tower_cli.get_resource('credential').get(workflow_node['credential'])['name']
+
+        if 'inventory' in workflow_node and workflow_node['inventory']:
+            node_to_add['inventory'] = tower_cli.get_resource('inventory').get(workflow_node['inventory'])['name']
+
+        # Finally copy over the different node types
+        for node_type in tower_cli.get_resource('workflow').workflow_node_types:
+            if node_type in workflow_node:
+                node_to_add[node_type] = workflow_node[node_type]
+
+        workflow_nodes_extracted.append(node_to_add)
+
+    # Finally we need to resolve all of the node IDs in the different types
+    for workflow_node in workflow_nodes_extracted:
+        for node_type in tower_cli.get_resource('workflow').workflow_node_types:
+            # Resolved nodes will be the resolved node names instead of IDs
+            resolved_nodes = []
+            for a_node_id in workflow_node[node_type]:
+                # If we found a node that does not resolve raise an exception
+                if a_node_id not in workflow_node_to_name_mapping:
+                    raise TowerCLIError("Workflow export exception: unable to resolve node {} from {}".format(
+                        a_node_id, asset['name'])
+                    )
+
+                # Add the new node to the list of resolved node
+                resolved_nodes.append(workflow_node_to_name_mapping[a_node_id])
+
+            # Put the resolved nodes back into the object
+            workflow_node[node_type] = resolved_nodes
+
+    return workflow_nodes_extracted
+#
```

Comment:
you have a lot of these stray `#` characters. Can we clean those up before the merge?

Reply:
Yea, we can whack them. Thats a personal style I use just to denote the end of a def. I'll update and push later today.